File: src/test/java/dev/retrotv/crypt/owe/sha/SHA256Test.java
package dev.retrotv.crypt.owe.sha;

import dev.retrotv.crypt.owe.OWETest;
import dev.retrotv.enums.HashAlgorithm;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

public class SHA256Test extends OWETest {

    @Test
    @DisplayName("SHA256 File hash 테스트")
    void fileHashTest() throws Exception {
        fileHashTest(HashAlgorithm.SHA256);
    }

    @Test
    @DisplayName("SHA256 File hash matches 테스트")
    void fileHashMatchesTest() throws Exception {
        fileHashMatchesTest(new SHA256(), HashAlgorithm.SHA256);
    }

    @Test
    @DisplayName("SHA256 File and File matches 테스트")
    void fileMatchesTest() throws Exception {
        fileMatchesTest(new SHA256());
    }

    @Test
    @DisplayName("SHA256 password encode 테스트")
    void passwordEncrypt() {
        passwordEncryptAndMatchesTest(new SHA256());
    }
}

File: src/test/kotlin/dev/retrotv/crypto/owe/hash/sha/SHA256Test.kt
package dev.retrotv.crypto.owe.hash.sha

import dev.retrotv.crypto.owe.OWETest
import dev.retrotv.enums.Algorithm
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

class SHA256Test : OWETest() {
    @Test
    @DisplayName("SHA256 File hash 테스트")
    @Throws(Exception::class)
    fun fileHashTest() {
        fileHashTest(Algorithm.Hash.SHA256)
    }

    @Test
    @DisplayName("SHA256 File hash matches 테스트")
    @Throws(Exception::class)
    fun fileHashMatchesTest() {
        fileHashMatchesTest(SHA256(), Algorithm.Hash.SHA256)
    }

    @Test
    @DisplayName("SHA256 password encode 테스트")
    fun passwordEncrypt() {
        passwordEncryptAndMatchesTest(SHA256())
    }
}
